module/VuFind/src/VuFind/View/Helper/Root/IdentifierLinker.php

```diff
@@ -97,41 +101,58 @@ public function __invoke($driver, $area)
     }
 
     /**
-     * Public method to render the OpenURL template
+     * Get all available identifiers.
      *
-     * @param bool $imagebased Indicates if an image based link
-     * should be displayed or not (null for system default)
+     * @return array
+     */
+    protected function getIdentifiers(): array
+    {
+        $ids = [];
+        if (in_array('doi', $this->supportedIdentifiers)) {
+            $ids['doi'] = $this->recordDriver->tryMethod('getCleanDOI');
```

Comment:
I believe those getters can also return `false`. Would it be better to add them only if a string was returned?